.github/workflows/docker-push.yml

```diff
@@ -0,0 +1,67 @@
+name: Docker Push
+
+on:
+  push:
+    branches: [ "main" ]
+    tags: [ "v*.*.*" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
```yaml
      - name: Get changed files
        id: changed-files
        uses: tj-actions/changed-files@v44
      - name: Check for VERSION file change
        id: version_changed
        env:
          ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
        run: |
          echo "changed=false" >> $GITHUB_ENV
          if echo "${ALL_CHANGED_FILES}" | grep -q 'VERSION'; then
            echo "changed=true" >> $GITHUB_ENV
          fi
      - name: Build
        if: env.changed == 'true'
        run: |
          python -m build
      - name: Publish package distributions to PyPI
        if: env.changed == 'true'
        uses: pypa/gh-action-pypi-publish@release/v1
```


Here is the example of `publish.yml` to pypi.
We deploy our version when the `VERSION` is changed. 
I think the docker also build and push to the docker hub when the `VERSION` is changed.